lib/src/git_backend.rs

```diff
@@ -756,13 +775,8 @@ mod tests {
         let commit_id1 = store.write_commit(&commit1).unwrap();
         let mut commit2 = commit1;
         commit2.predecessors.push(commit_id1.clone());
-        let expected_error_message = format!("Git commit '{}' already exists", commit_id1.hex());
-        match store.write_commit(&commit2) {
-            Ok(_) => {
-                panic!("expectedly successfully wrote two commits with the same git commit object")
-            }
-            Err(BackendError::Other(message)) if message.contains(&expected_error_message) => {}
-            Err(err) => panic!("unexpected error: {err:?}"),
-        };
+        // `write_commit` should prevent the ids from being the same by changing the
+        // committer timestamp of the commit it actually writes.
+        assert_ne!(store.write_commit(&commit2).unwrap(), commit_id1);
```

Comment:
It's curious that the underflow caused by subtracting a second from the 0 timestamp doesn't cause any problem here. I expected Rust to panic.

I think it's safe to assume the real timestamp is far from 0, but we could check and panic explicitly if we wanted.

I could also change the timestamp in the test, but I found it interesting that the code works even in this case.